apps/dashboard/app/views/batch_connect/sessions/connections/_native_vnc.html.erb

```diff
@@ -1,39 +1,56 @@
+<%# Random port from 10000 - 65535 deterministically generated %>
+<% localport = Digest::MD5.hexdigest("#{connect.host}:#{connect.port}").to_i(16) % 55535 + 10000 %>
+
 <% if browser.platform == :windows %>
```

Comment:
For windows, move the block into a partial like _native_vnc.windows.html.erb so we can override it with a custom one in /etc/ood/config/... for our deployment. We still need to support OSC Connect at OSC.